stress/client/main.go

```diff
@@ -0,0 +1,292 @@
+		default:
+			return nil, fmt.Errorf("unknown test type: %s", temp[0])
+		}
+		testCases[i].name = temp[0]
+		w, err := strconv.Atoi(temp[1])
+		if err != nil {
+			return nil, fmt.Errorf("%v", err)
+		}
+		testCases[i].weight = w
+	}
+	return testCases, nil
+}
+
+// weightedRandomTestSelector defines a weighted random selector for test case types.
+type weightedRandomTestSelector struct {
+	tests       []testCaseWithWeight
+	totalWeight int
+}
+
+// newWeightedRandomTestSelector constructs a weightedRandomTestSelector with the given list of testCaseWithWeight.
+func newWeightedRandomTestSelector(tests []testCaseWithWeight) *weightedRandomTestSelector {
+	var totalWeight int
+	for _, t := range tests {
+		totalWeight += t.weight
+	}
+	rand.Seed(time.Now().UnixNano())
+	return &weightedRandomTestSelector{tests, totalWeight}
+}
+
+func (selector weightedRandomTestSelector) getNextTest() (string, error) {
+	random := rand.Intn(selector.totalWeight)
+	var weightSofar int
+	for _, test := range selector.tests {
+		weightSofar += test.weight
+		if random < weightSofar {
+			return test.name, nil
+		}
+	}
+	return "", fmt.Errorf("no test case selected by weightedRandomTestSelector")
+}
+
+// gauge defines type for gauge.
+type gauge struct {
+	mutex sync.RWMutex
+	val   int64
+}
+
+// Set updates the gauge value
+func (g *gauge) set(v int64) {
+	g.mutex.Lock()
+	defer g.mutex.Unlock()
+	g.val = v
+}
+
+func (g *gauge) get() int64 {
+	g.mutex.RLock()
+	defer g.mutex.RUnlock()
+	return g.val
+}
+
+// Server implements metrics server functions.
+type server struct {
+	mutex  sync.RWMutex
+	gauges map[string]*gauge
+}
+
+// newMetricsServer returns a new metrics server.
+func newMetricsServer() *server {
+	return &server{gauges: make(map[string]*gauge)}
+}
+
+// GetAllGauges returns all gauges.
+func (s *server) GetAllGauges(in *metricspb.EmptyMessage, stream metricspb.MetricsService_GetAllGaugesServer) error {
+	s.mutex.RLock()
+	defer s.mutex.RUnlock()
+
+	for name, gauge := range s.gauges {
+		err := stream.Send(&metricspb.GaugeResponse{Name: name, Value: &metricspb.GaugeResponse_LongValue{gauge.get()}})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// GetGauge returns the gauge for the given name.
+func (s *server) GetGauge(ctx context.Context, in *metricspb.GaugeRequest) (*metricspb.GaugeResponse, error) {
+	s.mutex.RLock()
+	defer s.mutex.RUnlock()
+
+	if g, ok := s.gauges[in.Name]; ok {
+		return &metricspb.GaugeResponse{Name: in.Name, Value: &metricspb.GaugeResponse_LongValue{g.get()}}, nil
+	}
+	return nil, grpc.Errorf(codes.InvalidArgument, "gauge with name %s not found", in.Name)
+}
+
+// CreateGauge creates a guage using the given name in metrics server
+func (s *server) createGauge(name string) (*gauge, error) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+
+	grpclog.Printf("create gauge: %s", name)
+	if _, ok := s.gauges[name]; ok {
+		// gauge already exists.
+		return nil, fmt.Errorf("gauge %s already exists", name)
```

Comment:
you can panic here directly to simplify the function since you log.Fatalf on the caller side anyways.